### PR TITLE
custom llm module

### PR DESCRIPTION
**Pull request by Matt-Carey**
move completion calls to new module
add settings for easy api key and other repetitive config refactor streaming

### CODE REVIEW

File: InworldAI/Source/InworldAILLM/Private/Completion/InworldLLMCompletionAsyncActionBase.cpp [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        UE_LOG(LogInworldAILLM, Error, TEXT("Invalid Inworld Studio response. code=%d error=%s"), ResponseCode, *FullResponseString);
+        HandleComplete(false);
+    }
+
+    SetReadyToDestroy();
+}
+
+void UInworldLLMCompletionAsyncActionBase::HandleNextResponseChunk(FHttpResponsePtr Response)
+{
+    const FString NextResponseString = Response->GetContentAsString().RightChop(ProcessedContentLength);
```
Comment (wuharvey) [inline InworldAI/Source/InworldAILLM/Private/Completion/InworldLLMCompletionAsyncActionBase.cpp:106]: `RightChop` 🔪

File: InworldAI/Source/InworldAILLM/Public/Completion/InworldLLMCompletionAsyncActionBase.h [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+protected:
+	virtual FString GetCompletionType() const PURE_VIRTUAL(UInworldLLMCompletionAsyncActionBase::GetCompletionType, return FString{};)
+
+	virtual bool GetRequestJson(TSharedPtr<FJsonObject>&RequestJson) const PURE_VIRTUAL(UInworldLLMCompletionAsyncActionBase::GetRequestJson, return bool{};)
+	virtual bool HandleResponseJson(const TSharedPtr<FJsonObject>& ResponseJson) PURE_VIRTUAL(UInworldLLMCompletionAsyncActionBase::HandleResponse, return bool{};)
+
+	virtual void HandleComplete(bool bSuccess) PURE_VIRTUAL(UInworldLLMCompletionAsyncActionBase::HandleComplete)
+
+private:
+#if ENGINE_MAJOR_VERSION >= 5 && ENGINE_MINOR_VERSION > 3
```
Comment (wuharvey) [inline InworldAI/Source/InworldAILLM/Public/Completion/InworldLLMCompletionAsyncActionBase.h:55]: Oooo nice catch

**Review by wuharvey — APPROVED** [review]
Looks good to me, thank you for making these changes!

**Review by harvey-inworld — APPROVED** [review]
I am also Harvey and I approve this PR

**Pull request merged**